cpp/src/DataStorm/NodeI.cpp

```diff
@@ -315,11 +366,29 @@ NodeI::createSubscriberSession(
             _proxy,
             nullptr,
             [self = shared_from_this(), session, subscriber](exception_ptr ex)
-            { self->removePublisherSession(subscriber, session, ex); });
+            {
+                if (session)
+                {
+                    self->retryPublisherSessionCreation(subscriber, session, ex);
+                }
+                // Else node is shutting down, or the session was established by a concurrent call.
```

Comment:
The logic here remains unchanged. The `if (session)` check corresponds to the previous `if (session && ...)` that was inside removePublisherSession.

This refactor makes it clearer that we do not retry if the initial call to `initiateCreateSessionAsync` fails, as the retry logic is handled within the Session servant class, and at this point, a Session servant has not yet been created.